tests/testsuite/artifact_dep.rs

```diff
@@ -425,6 +425,85 @@ fn features_are_not_unified_among_lib_and_bin_dep_of_different_target() {
         .run();
 }
 
+#[cargo_test]
+fn feature_resolution_works_for_cfg_target_specification() {
+    if cross_compile::disabled() {
+        return;
+    }
+    let target = cross_compile::alternate();
+    let target_arch = cross_compile::alternate_arch();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            &r#"
+                [project]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+                resolver = "2"
+
+                [dependencies.d1]
+                path = "d1"
+                artifact = "bin"
+                target = "$TARGET"
+            "#
+            .replace("$TARGET", target),
+        )
+        .file(
+            "src/main.rs",
+            r#"
+                fn main() {
+                    let _b = include_bytes!(env!("CARGO_BIN_FILE_D1"));
+                }
+            "#,
+        )
+        .file(
+            "d1/Cargo.toml",
+            &r#"
+                [package]
+                name = "d1"
+                version = "0.0.1"
+                authors = []
+
+                [target.'cfg(target_arch = "$ARCH")'.dependencies]
```

Comment:
This should be something like the following:

```suggestion
                [target.'$TARGET'.dependencies]
```

On macOS, both the host and alt target arch are x86_64.

Reply:
Thanks for the suggestion, it's applied in 4e847a2c11b56bbc9e376febfa6af009d3335b16 as part of following up on the above comment.